google/gax/api_callable.py

```diff
@@ -417,7 +439,9 @@ def base_caller(api_call, _, *args):
 
     def inner(request, options=None):
         """Invoke with the actual settings."""
-        this_settings = settings.merge(options)
+        this_options = _merge_options_metadata(options, settings)
```

Comment:
I still disagree. I don't see any reason `_CallSettings` shouldn't know that the metadata field is special. I also don't think putting this behavior in `settings.merge` is dangerous -- if anything, it seems counterintuitive to me that you *wouldn't* get this behavior if you tried to merge an `Options` into a `Settings` using the method on the settings object.

If the other reviewers are fine with this, it's not something I feel strongly enough about to hold up the merge for.